src/Sowp/NewsModuleBundle/Controller/CollectionController.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Sowp\NewsModuleBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Sowp\NewsModuleBundle\Entity\Collection;
+use Sowp\NewsModuleBundle\Form\addCollectionForm as addForm;
+
+/**
+ * Collection controller.
+ *
+ * @Route("/kolekcje")
+ */
+class CollectionController extends Controller
+{
+    /**
+     * query collections for Select2Entity
+     * @param Request $request
+     *
+     * @Route("/query", name="collection_query_select2")
+     * @Method("GET")
+     */
+    public function queryAction(Request $request)
+    {
+
+        $collections = [[
+                'id' => 1,
+                'text' => '$c->getTitle()'
+            ],
+[
+                'id' => 2,
+                'text' => '$c->getTitle2()'
+            ],
+            ];
+        return new Response(json_encode($collections));
+    }
+
+    /**
+     * add new collection entry.
+     *
+     * @Route("/dodaj", name="addCollection")
+     * @Method({"GET","POST"})
+     */
+    public function addAction(Request $req)
+    {
+        $collection = new Collection();
+        $form = $this->createForm(addForm::class, $collection);
+        $form->handleRequest($req);
+
+        if ($form->isSubmitted()) {
+            if ($form->isValid()) {
+                $em = $this->getDoctrine()->getManager();
+                $em->persist($collection);
+                $em->flush();
+                $this->container->get('session')->getFlashBag()->add('notice', 'Zapisano');
+            } else {
+                $this->container->get('session')->getFlashBag()->add('error', 'Wystąpił błąd');
+            }
+        }
+
+        return $this->render('collection/add.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * add new collection entry.
+     *
+     * @Route("/edytuj/{id}", name="editCollection")
+     * @Method({"GET","POST"})
+     */
+    public function editAction(Request $req, Collection $col)
+    {
+    }
+
+    /**
+     * Lists all Collection entities.
+     *
+     * @Route("/", name="kolekcja_index")
+     * @Method("GET")
+     */
+    public function indexAction()
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $collections = $em->getRepository('Sowp:NewsModuleBundle:Collection')->findAll();
```

Comment:
Ja osobiście to lubie korzystać z skrótu z PHP składni ``::class``. Wtedy piszę ````Collection::class```` i jest sprawniej, bo IDE podpowiada.